packages/terra-form-validation/examples/index-examples/DefaultFormValidation.jsx

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable class-methods-use-this */
+
+import React from 'react';
+import { Form, Field } from 'react-final-form';
+import InputField from 'terra-form-input/lib/InputField';
+import TextareaField from 'terra-form-textarea/lib/TextareaField';
+import Button from 'terra-button';
+import Spacer from 'terra-spacer';
+
+const validateUniqueUser = async (name) => {
+  const response = new Promise((resolve) => {
+    if (name !== 'TerraUser') {
+      return resolve('');
+    }
+
+    return resolve('Name is Unavailable');
+  });
+
+  await response;
+  return response;
+};
+
+export default class MainEntry extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+    this.submitForm = this.submitForm.bind(this);
+    this.renderForm = this.renderForm.bind(this);
+  }
+
+  submitForm(values) {
+    this.setState({
+      submittedValues: values,
+    });
+  }
+
+  renderForm({ handleSubmit, pristine, invalid }) {
```

Comment:
Let's switch validation on this to be onSubmit rather than onInput for this default example, I believe onSubmit is the more common use case UX intends. cc/ @neilpfeiffer